sealir-tutorials/ch06_1_typeinfer_ifelse.py

```diff
@@ -0,0 +1,1580 @@
+# ---
```

Comment:
This file contains many duplications from ch4. All these can just be imported. This will shortened this file a lot. In the end, this file should mainly be the new Backend with the MLIR logic